lib/packwerk/application_load_paths.rb

```diff
@@ -18,36 +18,30 @@ def extract_relevant_paths(root, environment)
         relative_path_strings(relevant_paths)
       end
 
-      sig { returns(T::Array[String]) }
+      sig { returns(T::Hash[String, Module]) }
       def extract_application_autoload_paths
-        Rails.application.railties
-          .select { |railtie| railtie.is_a?(Rails::Engine) }
-          .push(Rails.application)
-          .flat_map do |engine|
-            paths = (engine.config.autoload_paths + engine.config.eager_load_paths + engine.config.autoload_once_paths)
-            paths.map(&:to_s).uniq
-          end
+        Rails.autoloaders.inject({}) do |h, loader|
+          h.merge(loader.root_dirs)
+        end
```

Comment:
So it turns out this doesn't work in Rails 6.0 unless `Rails.autoloaders.zeitwerk_enabled?` returns `true`. If it is false, `Rails.autoloaders` will be an empty enumerable.

So now we have to choose one of:

* For all the places that we stub out `config.autoload_paths` and friends we add matching tests to cover when zeitwerk is both enabled and disabled.
* Explicitly add the requirement that packwerk is only compatible with Rails applications which are zeitwerk enabled. This is the case for Rails 6.x codebases that have `Rails.autoloaders.zeitwerk_enabled? == true` or Rails 7.0+ codebases.

Personally, based on our discussions about adding Zeitwerk as a dev dependency in the [test in this diff](https://github.com/Shopify/packwerk/pull/190/files#r840817911) I feel we could pretty safely say that Packwerk requires `zeitwerk_enabled?` to be true. This would retain Rails 6.x support without requiring us to duplicate a whole bunch of test.

As ["Classic mode"](https://github.com/rails/rails/commit/0d523d83657ce7066f25d87f6f094e804590e1e8) isn't coming back, and Packwerk is based on the conventions of Zeitwerk, unless this causes problems for any known users, I'd suggest we go with the addition of this "soft" requirement.

Reply:
Maybe we should make it more obvious, but packwerk already officially requires zeitwerk to be enabled https://github.com/Shopify/packwerk#prerequisites

I'll accept a PR to reword that to make it more obvious